iron-menu-behavior.html

```diff
@@ -148,7 +148,12 @@
         var item = this.items[(curFocusIndex - i + length) % length];
         if (!item.hasAttribute('disabled')) {
           this._setFocusedItem(item);
-          return;
+
+          // Focus might not have worked, if the element was hidden or not
+          // focusable. In that case, try again.
+          if (document.activeElement == item) {
```

Comment:
don't we want to check that `this.shadowRoot.activeItem == item`?  (or some `Polymer.dom()`-y equivalent to work with shady DOM)?

I think `document.activeElement` is the first externally visible shadow root that contains the active element.

```html
  <x-foo>
    <x-bar>
      <x-baz>
        <input type="search" autofocus>
```

`document.activeElement` would be the top `<x-foo>`, I believe.